### PR TITLE
Feat/test cases

### PR DESCRIPTION
**Pull request by Hammad887**
Added table tests for functions.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+vendor/
+tmp/
+.env
+.vscode
```
Comment (shahzadhaider1) [inline .gitignore]: blank line at the end

File: cmd/ham_server/main.go [hunk 2]
```diff
@@ -32,11 +32,16 @@ func main() {
...
+	}()
 
 	server.ConfigureAPI()
 
 	if err := server.Serve(); err != nil {
-		log.Fatalln(err)
+		log.Println(err)
+		panic("Error occurred")
 	}
 }
```
Comment (shahzadhaider1) [inline cmd/ham_server/main.go]: `ham_server` in `cmd/ham_server/main.go` is not correct. 
should be meaningful and relevant name, like: `cmd/chat_server/main.go` or `cmd/chat_app_server/main.go`

File: db/mysql/ops_test.go [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/Hammad887/chat-app/db"
+	domain "github.com/Hammad887/chat-app/models"
+	"github.com/google/uuid"
```
Comment (shahzadhaider1) [inline db/mysql/ops_test.go:11]: please sort imports

**Review by shahzadhaider1 — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (Hammad887) [reply .gitignore]: Added.

File: cmd/ham_server/main.go (hunk 2 above)
Comment (Hammad887) [reply cmd/ham_server/main.go]: Changed

File: db/mysql/ops_test.go (hunk 3 above)
Comment (Hammad887) [reply db/mysql/ops_test.go:11]: Sorted

File: handlers/sendMessageHandler.go [hunk 4]
```diff
@@ -55,7 +55,15 @@ func (r *sendMessage) Handle(params service.SendMessageParams) middleware.Respon
...
 
+<<<<<<< HEAD
+<<<<<<< HEAD
 // SendMessageHandler returns a handler for sending messages.
+=======
+// SendMessageHandler creates and returns a handler for sending messages using the provided runtime.
+>>>>>>> 8e32e52 (linting)
+=======
+// SendMessageHandler creates and returns a handler for sending messages using the provided runtime.
+>>>>>>> master
```
Comment (shahzaibaziz) [inline handlers/sendMessageHandler.go]: resolve these conflicts

File: handlers/registerHandler.go [hunk 5]
```diff
@@ -53,7 +54,15 @@ func (r *registerUser) Handle(params service.RegisterUserParams) middleware.Resp
...
 
+<<<<<<< HEAD
+<<<<<<< HEAD
 // RegisterUserHandler returns a handler that manages user resistration.
+=======
+// RegisterUserHandler creates and returns a handler for registering users using the provided runtime.
+>>>>>>> 8e32e52 (linting)
+=======
+// RegisterUserHandler creates and returns a handler for registering users using the provided runtime.
+>>>>>>> master
```
Comment (shahzaibaziz) [inline handlers/registerHandler.go]: conflicts

File: handlers/logoutHandler.go [hunk 6]
```diff
@@ -40,7 +40,15 @@ func (r *logoutUser) Handle(params service.LogoutParams) middleware.Responder {
...
 
+<<<<<<< HEAD
+<<<<<<< HEAD
 // LogoutUserHandler returns a handler that manages user logout.
+=======
+// LogoutUserHandler creates and returns a handler for logging out users using the provided runtime.
+>>>>>>> 8e32e52 (linting)
+=======
+// LogoutUserHandler creates and returns a handler for logging out users using the provided runtime.
+>>>>>>> master
```
Comment (shahzaibaziz) [inline handlers/logoutHandler.go]: conflicts

File: handlers/chatRoomHandler.go [hunk 7]
```diff
@@ -41,10 +41,17 @@ func (r *getChatroom) Handle(params service.GetChatroomParams) middleware.Respon
...
 
+<<<<<<< HEAD
+<<<<<<< HEAD
 // GetChatroomHandler returns a handler for retrieving a chat room.
+=======
+// GetChatroomHandler creates and returns a handler for fetching chat rooms using the provided runtime.
+>>>>>>> 8e32e52 (linting)
+=======
+// GetChatroomHandler creates and returns a handler for fetching chat rooms using the provided runtime.
+>>>>>>> master
```
Comment (shahzaibaziz) [inline handlers/chatRoomHandler.go]: need a clean merge

**Pull request merged**